os/arch/arm/src/imxrt/imxrt_mtd_flash.c

```diff
@@ -405,9 +404,9 @@ static int imxrt_erase(FAR struct mtd_dev_s *dev, off_t startsector, size_t nsec
 		imxrt_sectorerase(priv, startsector + sectors_erased);
 		sectors_erased++;
 	}
-	#endif
-	
-	return (int)sectors_erased;
+	fvdbg("erased sectors = %d\n", sectors_erased);
```

Comment:
I think you still confused, MTD is to use flash like a hard disk.
Page is basic unit size of write, Block basic unit size of erase.
So 'block' is correct for imxrt_erase, because erase is working block by block.
Then, I think you can ask below
why imxrt_sectorerase is here? Why it has 'sector' as a first parameter?
It because some device has huge size of block.
In this case, erase will take bunch of time also it's not good option for tiny device.
So, some device provides 'subsector of eraseblock' as a basic unit of erase.
so in fact, start_sector of imxrt_sectorerase is start block.....

Reply:
Yes, I understood what you want to talk.
As you know, the terms sector and block are used interchangeably and meanings of two terms are different from device by device..
Here I used the term "sector" since sector is used to denote an erasable block in imxrt_dev_s.
Also in http://nuttx.org/doku.php?id=wiki:vfs:using-smartfs , it is described that Sector: Same as Erase Block ...